test/resources/knativeserving.go

```diff
@@ -21,68 +21,88 @@ package resources
+		return inState(lastState, err)
 	})
 
 	if waitErr != nil {
-		return lastState, errors.Wrapf(waitErr, "knativeserving %q is not in desired state, got: %+v", name, lastState)
+		return lastState, errors.Wrapf(waitErr, "knativeserving %s is not in desired state, got: %+v", name, lastState)
 	}
 	return lastState, nil
 }
 
-// IsKnativeServingReady will check the status conditions of the KnativeServing and return true if the KnativeServing is ready.
-func IsKnativeServingReady(s *v1alpha1.KnativeServing) (bool, error) {
-	return s.Status.IsReady(), nil
+// CreateKnativeServing creates a KnativeServing with the name names.KnativeServing under the namespace names.Namespace.
+func CreateKnativeServing(clients servingv1alpha1.KnativeServingInterface, names test.ResourceNames) (*v1alpha1.KnativeServing, error) {
+	ks := &v1alpha1.KnativeServing{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: names.KnativeServing,
+			Namespace: names.Namespace,
+		},
+	}
+	svc, err := clients.Create(ks)
+	return svc, err
 }
 
-// WaitForDeploymentAvailable polls the status of the deployment called name
+// WaitForKnativeServingState polls the status of the KnativeServing called name
 // from client every `interval` until `inState` returns `true` indicating it
 // is done, returns an error or timeout.
-func WaitForDeploymentAvailable(clients *test.Clients, name, namespace string, inState func(s *v1.Deployment) (bool, error)) (*v1.Deployment, error) {
-	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForKnativeServingState/%s/%s", name,
-		"DeploymentIsAvailable"))
+func WaitForKnativeServingState(client servingv1alpha1.KnativeServingInterface, name string, inState func(s *v1alpha1.KnativeServing,
+	err error) (bool, error)) (*v1alpha1.KnativeServing, error) {
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForKnativeServingState/%s/%s", name, "KnativeServingIsReady"))
 	defer span.End()
-	var dep *v1.Deployment
 
-	waitErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
-		dep, _ = clients.KubeClient.Kube.AppsV1().Deployments(namespace).Get(name, va1.GetOptions{})
-		return inState(dep)
+	var lastState *v1alpha1.KnativeServing
+	waitErr := wait.PollImmediate(Interval, Timeout, func() (bool, error) {
+		lastState, err := client.Get(name, metav1.GetOptions{})
+		return inState(lastState, err)
 	})
 
 	if waitErr != nil {
-		return dep, errors.Wrapf(waitErr, "Deployment %q is not in desired status for the condition type Available,"+
-			"got: %+q; want %+q", name, getDeploymentStatus(dep), "True")
+		return lastState, errors.Wrapf(waitErr, "knativeserving %q is not in desired state, got: %+v", name, lastState)
 	}
+	return lastState, nil
+}
+
+// IsKnativeServingReady will check the status conditions of the KnativeServing and return true if the KnativeServing is ready.
+func IsKnativeServingReady(s *v1alpha1.KnativeServing, err error) (bool, error) {
+	return s.Status.IsReady(), nil
```

Comment:
I feel like these helpers should surface unexpected errors, i.e. return

```golang
s.Status.IsReady(), err
```

here and

```
if !apierrs.IsNotFound(err) {
  return false, err
}
return true, nil
```

below.

Reply:
No need of the func IsKnativeServingDeleted anymore, so I can remove it. Will change IsKnativeServingReady